common/beerocks/bwl/nl80211/sta_wlan_hal_nl80211.cpp

```diff
@@ -59,7 +90,21 @@ bool sta_wlan_hal_nl80211::unassoc_rssi_measurement(const std::string &mac, int
     return true;
 }
 
-bool sta_wlan_hal_nl80211::is_connected() { return true; }
+bool sta_wlan_hal_nl80211::is_connected()
+{
+    ConnectionStatus connection_status;
+    int retries = 10;
+    while (!read_status(connection_status)) {
+        --retries;
+        LOG(ERROR) << "is_connected - read_status failed, retry:" << 10 - retries;
```

Comment:
Nitpick: Retries can be in another direction, like ``retries++ ``. And `10 - retries` can be changed on `retries`.